cql3/column_condition.cc

```diff
@@ -123,82 +78,15 @@ bool column_condition::applies_to(const expr::evaluation_inputs& inputs) const {
 
 lw_shared_ptr<column_condition>
 column_condition::raw::prepare(data_dictionary::database db, const sstring& keyspace, const schema& schema) const {
-    auto id = _lhs->prepare_column_identifier(schema);
-    const column_definition* def = get_column_definition(schema, *id);
-    if (!def) {
-        throw exceptions::invalid_request_exception(format("Unknown identifier {}", *id));
-    }
-
-    if (def->is_primary_key()) {
-        throw exceptions::invalid_request_exception(format("PRIMARY KEY column '{}' cannot have IF conditions", *id));
-    }
-
-    auto& receiver = *def;
+    auto prepared = expr::prepare_expression(_expr, db, keyspace, &schema, make_lw_shared<column_specification>("", "", make_shared<column_identifier>("IF condition", true), boolean_type));
```

Comment:
We could get the kespace and table name for `column_specification` constructor from the `schema`.